flake.nix

```diff
@@ -24,372 +23,81 @@
-                  echo "ERROR: yosys version in Makefile isn't equivalent to version of the nix package (allegedly ${version}), failing."
-                  exit 1
-                fi
-              '';
-
-            checkTarget = "test";
-            doCheck = false;
-            checkInputs = [ verilog ];
-
-            # Internally, yosys knows to use the specified hardcoded ABCEXTERNAL binary.
-            # But other tools (like mcy or symbiyosys) can't know how yosys was built, so
-            # they just assume that 'yosys-abc' is available -- but it's not installed
-            # when using ABCEXTERNAL
-            #
-            # add a symlink to fake things so that both variants work the same way. this
-            # is also needed at build time for the test suite.
-            postBuild = "ln -sfv ${abc-verifier}/bin/abc ./yosys-abc";
-            postInstall = "ln -sfv ${abc-verifier}/bin/abc $out/bin/yosys-abc";
-
-            setupHook = ./yosys-setup-hook.sh;
-
-            passthru = let
-              withPlugins = plugins:
-                let
-                  paths = lib.closePropagation plugins;
-                  module_flags = with builtins;
-                    concatStringsSep " "
-                    (map (n: "--add-flags -m --add-flags ${n.plugin}") plugins);
-                in lib.appendToName "with-plugins" (lib.symlinkJoin {
-                  inherit (yosys) name;
-                  paths = paths ++ [ yosys ];
-                  nativeBuildInputs = [ makeWrapper ];
-                  postBuild = ''
-                    wrapProgram $out/bin/yosys \
-                      --set NIX_YOSYS_PLUGIN_DIRS $out/share/yosys/plugins \
-                      ${module_flags}
-                  '';
-                });
+            doCheck = true; # FIXME(ac): can we turn these back on?
 
+            passthru = {
+              inherit (prev) withPlugins;
               allPlugins = { ghdl = yosys-ghdl; };
-            in { inherit withPlugins allPlugins; };
-
-            meta = with lib; {
-              description = "Open RTL synthesis framework and tools";
-              homepage = "https://yosyshq.net/yosys/";
-              license = licenses.isc;
-              platforms = platforms.all;
-              maintainers = with maintainers; [ thoughtpolice ];
             };
-          };
-
-        nextpnr-xilinx = with final;
-          stdenv.mkDerivation rec {
-            pname = "nextpnr-xilinx";
-            version = "0.5.0";
-
-            srcs = [
-              (fetchgit {
-                url = "https://github.com/openXC7/nextpnr-xilinx";
-                rev = version;
-                fetchSubmodules = true;
-                deepClone = false;
-                hash = "sha256-mDYEmq3MW1kK9HeR4PyGmKQnAzpvlOf+H66o7QTFx3k=";
-                leaveDotGit = false;
-              })
-            ];
-
-            sourceRoot = "nextpnr-xilinx";
+          })).override { inherit abc-verifier; };
```

Comment:
Similar to abc-verifier, our changes consisted of using an override and only the minimal field changes required from the upstream derivation. We also need to override the `abc-verifier` input here so that we use the one defined above with the specific version

Reply:
ok